docs/getting_started/setting_up_an_api.md

```diff
@@ -0,0 +1,112 @@
+# Setting up an OPTIMADE API
+
+These notes describe how to set up and customize an OPTIMADE API based on the reference server in this package for some existing crystal structure data.
+
+To follow this guide, you will need to have a working development installation, as described in the [installation instructions](../INSTALL.md#full-development-installation).
+Complete examples of APIs that use this package are described in the [Use Cases](./use_cases.md) section.
+
+## Setting up the database
+
+The `optimade` reference server requires a data source per OPTIMADE entry type (`structures`, `references`, `links`).
+In the simplest case, these can be configured as named MongoDB collections with a defined MongoDB URI and database name (see below), but they can also be set up as custom subclasses of [`EntryCollection`][optimade.server.entry_collections.entry_collections.EntryCollection] that could simply read from a static file.
+In the reference server, these data sources, or collections, are created in the submodule for the corresponding routers/endpoints.
+
+Here, we shall use the built-in MongoDB collections for each entry type, by simply specifying the appropriate options in the [configuration](../configuration.md), namely [`"database_backend": "mongodb"`][optimade.server.config.ServerConfig.database_backend], [`"mongo_uri": "mongodb://localhost:27017"`][optimade.server.config.ServerConfig.mongo_uri], [`"mongo_database": "optimade"`][optimade.server.config.ServerConfig.mongo_database] and the collection names for each entry type ([`"structures_collection": "structures"`][optimade.server.config.ServerConfig.structures_collection] etc.).
+These notes will now assume that you have a MongoDB instance running and you have created a database that matches your [`"mongo_database"`][optimade.server.config.ServerConfig.mongo_database] config option.
+
+If you disable inserting test data (with the [`"insert_test_data": false`][optimade.server.config.ServerConfig.insert_test_data] configuration option), you can test your API/database connection by running the web server with `uvicorn optimade.main:app` and visiting the (hopefully empty) structures endpoint at `localhost:5000/v1/structures` (or your chosen base URL).
```

Comment:
I think the command to start the web server should be `uvicorn optimade.server.main:app`
Once I have finished setting up a local API I will make a pull request but if that is not fast enough someone else can do this already.

Reply:
When I start the program with uvicorn the data is shared at `localhost:8000/v1/structures` even if I specify `"base_url": "http://localhost:5000"`, in `.optimade.json`.

Reply:
You also need to set it in the call to `uvicorn`, e.g. `uvicorn --port 5000` as we do in the `run.sh` script. I can clarify this in the text. The `base_url` config option is just used to set appropriate links internally.